### DISCUSSION TITLE
can I have a js version of the file schema.prisma?

### DISCUSSION THREAD

**Discussion by Rahmo** [prisma/prisma#5824] [Q&A] [answered]
Hi all I have a question regarding the file schema.prisma I want it in js so as I have to make calls from env of the database url we don't store database url in repo

**Comment by joshuaavalon** [comment, marked as answer]
In `schema.prisma`, you can use environment variables like

```prisma
datasource db {
  provider = "postgresql"
  url      = env("DATABASE_URL")
}
```

You can also override it in the client with
```typescript
import { PrismaClient } from "@prisma/client";
const url = process.env.DATABASE_URL;
const client = new PrismaClient({ datasources: { db: { url } } });
```

Note that Prisma command line also support reading from `.env`.

**Reply by Rahmo** [reply]
@joshuaavalon the problem im facing is that im getting the url link from a secure service that returns a promise 
we don't store sensitive data in repo
can I do something like this 
const url = await getUrl();

**Reply by joshuaavalon** [reply]
@Rahmo It does not matter and you can still do it like this.

```js
// script.js
const url = await getUrl();
console.log(url)
```

```bash
DATABASE_URL=$(node script.js)
prisma introspect
```

**Reply by Rahmo** [reply]
Thx @joshuaavalon 
I have trying the code override since my case is involving getting data from a promise its still return a promise to the context without it being resolved 

```
// context.js
const { PrismaClient } = require('@prisma/client')
const ssmUtil = require('../aws_ssm_util');
const { AppName } = require('../constants');

//api to get a secure connection string 
async function getUrl(){
    return await ssmUtil.getSSMParameter(paramName).then(res => {
      return res;
    }).catch(()=>{
      throw 'Unable to fetch connection string from SSM.';
    });
  }
}

const prisma = async function () {
  return new PrismaClient({
    datasources: {
      db: {
        url: await getUrl()
      }
    }
  });
}

const createContext =
  async function () {
    return {
      prisma: await prisma()
    }
  }

module.exports = {
  createContext
}

```
//in api.js
`
// apollo Grapghql
const server = new ApolloServer({ typeDefs, resolvers, context: createContext });
server.applyMiddleware({ app:router });
`
any thoughts

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
